### ISSUE TITLE
'Goto Directory' ignores hidden folders

### ISSUE THREAD

**Issue by cgaerter** [Microsoft/winfile#100]
I have set my (Windows 7) system to not display hidden files/folders, but since the path to the desktop starts with the hidden users folder in the system root, I cannot navigate to the desktop or to my personal files.

I suggest to either add an option to WinFile to display hidden files/folders without changing the system-wide setting or to enable the 'Goto Directory' feature to jump to folders below a hidden folder.

**Comment by craigwi** [comment]
Hidden and other types of directories can certainly be indexed.  In my use case, doing so resulted in too much noise.  But I think an option makes sense. 

We probably need to start an options dialog for the goto command since another obvious fix to make is to index other drives if desired.

**Comment by DBJDBJ** [comment]
IMHO including hidden folders will indeed make humongous amount of noise
other drives are it seems already indexed? let me check once more ...

Ok, I have C:, D: H: drives and their windows open , while on D: , if I do CTRL+G and type "u", "c:\users" is immediately shown ...

I close C: and H: windows and still same as before ..  I do CTRL+G and type "u", "c:\users" is immediately shown ...

I assume this was fixed before I joined the project :)

**Comment by DBJDBJ** [comment]
IMHO the "obvious fix" might be to search only the active drive ... 
right now wherever, on the disks, one is, only C: is indexed (true) and if user desired to "go to" the result current window changes to c: but the previously active drive stays as selected on the drives tool bar. 

I assume this would be easy to fix ...

HTH

**Comment by toiletflusher** [comment]
Old issue post/thread, i know, but you can enable hidden files and folders from winfile already. Even in the NT 4.0 version.
Click the View menu, Click on by file type...
Then it lets you choose to show it or not. it also fixes the problem that Goto Directory ignores hidden folders.

**Comment by JBensimon99** [comment]
Yes, showing hidden files and folders is an available option, BUT that setting is not preserved when saving settings for future launches of WinFile (which by the way are stored in "%AppData%\Roaming\Microsoft\Winfile\WINFILE.INI"). This should be an easy fix.

**Comment by malxau-msft** [comment]
I think everyone here is describing a slightly different thing.

- @JBensimon99 Can you describe more about the sequence of steps that cause the hidden display of files to not be saved? I verified that this is saved for me on application exit as well as via explicit save.
- @0penrc Note that the hidden setting is per-window but the goto cache is only stored once.  Using the per-window setting implies indexing everything and filtering the index when something is searched for. The goto cache isn't attempting to look at that today - it ignores hidden/system files unconditionally https://github.com/microsoft/winfile/blob/e3075d599a763c9ef86a0a612a83fd9b6976c243/src/wfgoto.cpp#L337 .
- I Totally agree with Craig that this would benefit from a configuration UI.  I started one before losing interest.  Note though that the UI is still distinct from a setting which determines what to index.

**Comment by JBensimon99** [comment]
> * @JBensimon99 Can you describe more about the sequence of steps that cause the hidden display of files to not be saved? I verified that this is saved for me on application exit as well as via explicit save.

You're right, I didn't realize the "show hidden" setting was window-specific -- I'd set it for one window, but looked at it after restarting WinFile while another window was active.  Thanks.
